happiness_app/templates/happiness_app/index.html

```diff
@@ -4,61 +4,74 @@
-                data-bs-slide="prev">
-                <span class="carousel-control-prev-icon" aria-hidden="true"></span>
-                <span class="visually-hidden">Previous</span>
-            </button>
-            <button class="carousel-control-next" type="button" data-bs-target="#carouselIndicators"
-                data-bs-slide="next">
-                <span class="carousel-control-next-icon" aria-hidden="true"></span>
-                <span class="visually-hidden">Next</span>
-            </button>
+  <div class="container">
+    <div id="carouselIndicators" class="carousel slide carousel-fade">
+      <div class="carousel-indicators">
+        {% for act in acts %}
+        <button type="button" data-bs-target="#carouselIndicators" data-bs-slide-to="{{ forloop.counter0 }}" 
+        {% if forloop.first %}class="active" aria-current="true" {% endif %}
+          aria-label="Slide {{ forloop.counter }}"></button>
+        {% endfor %}
+      </div>
+      <div class="carousel-inner">
+        {% for act in acts %}
+        <div class="carousel-item {% if forloop.first %}active{% endif %}">
+          <img src="{{ act.image.url }}" class="d-block w-100" alt="{{ act.name }}" />
+          <div class="carousel-caption d-none d-md-block">
+            <h5>Slide {{ forloop.counter }} label</h5>
+            <p>Some representative placeholder content for slide {{ forloop.counter }}.</p>
+          </div>
         </div>
+        {% endfor %}
+      </div>
+      <button class="carousel-control-prev" type="button" data-bs-target="#carouselIndicators" data-bs-slide="prev">
+        <span class="carousel-control-prev-icon" aria-hidden="true"></span>
+        <span class="visually-hidden">Previous</span>
+      </button>
+      <button class="carousel-control-next" type="button" data-bs-target="#carouselIndicators" data-bs-slide="next">
+        <span class="carousel-control-next-icon" aria-hidden="true"></span>
+        <span class="visually-hidden">Next</span>
+      </button>
     </div>
+  </div>
 </section>
 <section class="pb-5" id="acts-of-kindness-cards">
-    <div class="container py-5">
-        <div class="row">
-            {% for act in random_acts %}
-            <div class="col-md-3">
-                <div class="card" style="width: 18rem;">
-                    {% if act.image %}
-                    <img src="{{ act.image.url }}" class="card-img-top" alt="{{ act.name }}">
-                    {% endif %}
-                    <div class="card-body">
-                        <h5 class="card-title">{{ act.name }}</h5>
-                        <p class="card-text">{{ act.description }}</p>
-                        <a href="{% url 'act_detail' act.id %}" class="btn btn-primary">Choose this act of kindness</a>
-                    </div>
-                </div>
-            </div>
-            {% endfor %}
+  <div class="container py-5">
+    <div class="row overflow-auto flex-nowrap">
```

Comment:
Suggestion (Non-blocking): Instead of using `overflow-auto` here, you could use the Bootstrap carousel that is available in the documentation. The reason I suggest that is because it likely already has touch events built in so that user can easily "swipe" through on mobile. We could also make it so that the card takes up 100% on mobile.